regcomp.c

```diff
@@ -1213,6 +1213,32 @@ S_set_regex_pv(pTHX_ RExC_state_t *pRExC_state, REGEXP *Rx)
     SvCUR_set(Rx, p - RX_WRAPPED(Rx));
 }
 
+static void
+release_RExC_state(pTHX_ void *vstate) {
+    RExC_state_t *pRExC_state = (RExC_state_t *)vstate;
+
+    if (RExC_rx_sv)                                                     \
+        SvREFCNT_dec(RExC_rx_sv);                                         \
+    if (RExC_open_parens) {
```

Comment:
Aren't all of these `if` checks redundant? `free` should be a no-op on a NULL pointer.